pipeline-model-definition/src/test/java/org/jenkinsci/plugins/pipeline/modeldefinition/WhenStageTest.java

```diff
@@ -386,4 +398,14 @@ protected boolean shouldExamineAllBuilds(@Nonnull SCMHead head) {
             return false;
         }
     }
+
+    @TestExtension
```

Comment:
shouldn't you isolate this to only be active for `whenExprDurableTask`?

Reply:
Theoretically, we could do that, but I'm frankly fine with this being run on all the tests in here anyway.